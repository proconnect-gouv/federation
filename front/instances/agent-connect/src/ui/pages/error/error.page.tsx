import React from 'react';

import { ErrorComponent } from './error.component';
import { NotFoundComponent } from './not-found.component';

export const ErrorPage = React.memo(() => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const hasError = (window as any).appError !== undefined;
  if (hasError) {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return <ErrorComponent errors={(window as any).appError} />;
  }
  return <NotFoundComponent />;
});

ErrorPage.displayName = 'ErrorPage';
