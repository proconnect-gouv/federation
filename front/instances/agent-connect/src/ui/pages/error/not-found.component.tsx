import classnames from 'classnames';
import React from 'react';
import { useMediaQuery } from 'react-responsive';

import styles from './not-found.module.scss';

export const NotFoundComponent = React.memo(() => {
  const gtTablet = useMediaQuery({ query: '(min-width: 768px)' });
  return (
    <div
      className={classnames(styles.page, 'fr-m-auto', {
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-mt-4w': !gtTablet,
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-mt-8w': gtTablet,
      })}>
      <h1 className="text-center">404 - Not Found</h1>
    </div>
  );
});

NotFoundComponent.displayName = 'NotFoundComponent';
