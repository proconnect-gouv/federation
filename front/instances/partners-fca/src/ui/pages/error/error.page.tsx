import React from 'react';
import { Helmet } from 'react-helmet';

export const ErrorPage = React.memo(() => (
  <React.Fragment>
    <Helmet>
      <title>Partenaires AgentConnect - Erreur</title>
    </Helmet>
    <div className="fr-container text-center fr-mt-8w">
      <h1 className="text-center">Une erreur est survenue</h1>
      <h2 className="text-center">
        <span>Erreur</span>
      </h2>
      <p className="text-center">Ceci est une erreur</p>
    </div>
  </React.Fragment>
));

ErrorPage.displayName = 'ErrorPage';
