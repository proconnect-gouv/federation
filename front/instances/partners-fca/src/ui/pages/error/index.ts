/* istanbul ignore file */

// declarative file
export * from './error.page';
