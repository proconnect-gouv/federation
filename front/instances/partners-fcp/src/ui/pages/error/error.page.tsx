export const ErrorPage = () => (
  <div className="fr-container text-center fr-mt-8w">
    <h1 className="text-center">Une erreur est survenue</h1>
    <h2 className="text-center">
      <span>Erreur</span>
    </h2>
    <p className="text-center">Ceci est une erreur</p>
  </div>
);
