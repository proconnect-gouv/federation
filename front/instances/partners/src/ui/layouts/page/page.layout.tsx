import React from 'react';
import { Outlet } from 'react-router-dom';

export const PageLayout = React.memo(() => (
  <main className="fr-container fr-py-8v">
    <div className="fr-grid-row fr-grid-row--center">
      <Outlet />
    </div>
  </main>
));

PageLayout.displayName = 'PageLayout';
