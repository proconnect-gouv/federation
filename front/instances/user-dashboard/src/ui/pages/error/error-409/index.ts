export * from './error-409.component';
