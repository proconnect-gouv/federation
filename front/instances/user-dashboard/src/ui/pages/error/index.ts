/* istanbul ignore file */

// declarative file
export * from './error.page';
export * from './error-409';
export * from './error-generic';
