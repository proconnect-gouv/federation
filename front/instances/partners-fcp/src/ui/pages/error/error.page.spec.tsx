import { render } from '@testing-library/react';
import { Component } from 'react';
import { Helmet } from 'react-helmet';

import { ErrorPage } from './error.page';

jest.mock('react-helmet');

describe('ErrorPage', () => {
  it('should match the snapshot', () => {
    // when
    const { container } = render(<ErrorPage />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should set page title through react helmet', () => {
    // given
    jest.mocked(Helmet).mockImplementationOnce(({ children }) => children as unknown as Component);

    // when
    const { getByText } = render(<ErrorPage />);
    const element = getByText('Partenaires FCP - Erreur');

    // then
    expect(Helmet).toHaveBeenCalledTimes(1);
    expect(element).toBeInTheDocument();
  });
});
