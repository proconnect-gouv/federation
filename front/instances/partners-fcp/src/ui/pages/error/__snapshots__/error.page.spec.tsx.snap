// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should match the snapshot 1`] = `
<div>
  <div
    class="fr-container text-center fr-mt-8w"
  >
    <h1
      class="text-center"
    >
      Une erreur est survenue
    </h1>
    <h2
      class="text-center"
    >
      <span>
        Erreur
      </span>
    </h2>
    <p
      class="text-center"
    >
      Ceci est une erreur
    </p>
  </div>
</div>
`;
